mangaki/mangaki/wrappers/anilist.py

```diff
@@ -192,335 +200,314 @@ def anilist_id(self) -> int:
+            }
+        )
         data = r.json()
 
-        if not isinstance(data, list) and data.get('error'):
-            if is_no_results(data['error']):
-                return None
-            elif is_route_not_found(data['error']):
-                raise AniListException('"{}" API route does not exist'.format(datapage))
-            elif is_token_error(data['error']):
-                raise AniListException('token no longer valid or not found')
-            else:
-                raise AniListException(data['error'])
+        if data.get('errors'):
+            for error in data['errors']:
+                raise AniListException(error)
 
         r.raise_for_status()
-        return data
-
-    def list_seasonal_animes(self,
-                             *,
-                             only_airing: Optional[bool] = True,
-                             year: Optional[int] = None,
-                             season: Optional[str] = None) -> Generator[AniListEntry, None, None]:
-        if not year or not season:
-            now = datetime.now()
-            year = now.year
-            season = to_anime_season(now)
+        return data['data']
 
-        query_params = {'year': year, 'season': season, 'full_page': 'true'}
+    def get_work(self,
+                 search_id: Optional[int] = None,
+                 search_title: Optional[str] = None) -> AniListEntry:
+        """
+        Retrieve a work's information entry from AniList, given an ID or/and a title.
+        :param self: an AniList client
+        :param search_id: the ID of the work to look for
+        :param search_title: the title of the work to look for
+        :type self: AniList
+        :type search_id: Optional[int]
+        :type search_title: Optional[str]
+        :return: the entry that was looked for
+        :rtype: AniListEntry
+        """
 
-        if only_airing:
-            query_params.update({'status': AniListStatus.airing.value})
+        if search_id is None and search_title is None:
+            raise ValueError("Please provide an ID or a title")
 
-        data = self._request('browse/anime', query_params=query_params)
-        for anime_info in data:
-            yield AniListEntry(anime_info, AniListWorks.animes)
+        variables = {}
+        variables.update({'id': search_id} if search_id else {})
+        variables.update({'search': search_title} if search_title else {})
 
-    def get_work_by_id(self,
-                       worktype: AniListWorks,
-                       id: int) -> AniListRichEntry:
-        """
-        Search a work by ID on AniList and returns a rich entry if the ID exists,
-        or None if there are no results.
-        A rich entry has informations about characters, staff, studio and even
-        related works.
-        """
         data = self._request(
-            '{worktype}/{id}/page',
-            {'worktype': worktype.value, 'id': id}
+            query=read_graphql_query('work-info'),
```

Comment:
We could make `'work-info'` as a constant static string in the head of this file, so that we can easily change all occurrences of it.